src/lib/browse/CensusOutputAreaLayerControl.svelte

```diff
@@ -0,0 +1,244 @@
+    overwritePolygonLayer,
+  } from "../../maplibre_helpers";
+  import { map } from "../../stores";
+  import { ExternalLink, HelpButton, InteractiveLayer } from "../common";
+  import { Checkbox } from "../govuk";
+  import { colors } from "./colors";
+  import SequentialLegend from "./SequentialLegend.svelte";
+
+  let name = "census_output_areas";
+  let outlineLayer = `${name}-outline`;
+  let colorScale = colors.sequential_low_to_high;
+
+  // Mutually exclusive, like a radio button
+  let showHouseholdsWithoutCar = false;
+  let showAverageCars = false;
+  let showPopulationDensity = false;
+  let colorBy = "";
+  $: {
+    if (showHouseholdsWithoutCar) {
+      colorBy = "percent_households_without_car";
+    } else if (showAverageCars) {
+      colorBy = "average_cars_per_household";
+    } else if (showPopulationDensity) {
+      colorBy = "pop_density";
+    } else {
+      colorBy = "";
+    }
+    if (colorBy) {
+      $map.setPaintProperty(name, "fill-color", makeColorRamp());
+      // InteractiveLayer manages the polygon layer, but we also need to control the outline
+      $map.setLayoutProperty(outlineLayer, "visibility", "visible");
+    } else {
+      $map.setLayoutProperty(outlineLayer, "visibility", "none");
+    }
+  }
+
+  overwritePmtilesSource(
+    $map,
+    name,
+    `https://atip.uk/layers/v1/${name}.pmtiles`
+  );
+
+  overwritePolygonLayer($map, {
+    id: name,
+    source: name,
+    sourceLayer: name,
+    // Initially set to a dummy value
+    color: "black",
+    opacity: hoveredToggle(0.5, 0.7),
+  });
+  overwriteLineLayer($map, {
+    id: outlineLayer,
+    source: name,
+    sourceLayer: name,
+    color: "black",
+    width: 0.5,
+  });
+
+  function tooltip(feature: MapGeoJSONFeature): string {
+    let oa = feature.properties["OA21CD"];
+    let value = feature.properties[colorBy];
+    if (colorBy == "percent_households_without_car") {
+      return `<p>${value}% of households in ${oa} don't have a car</p>`;
+    } else if (colorBy == "average_cars_per_household") {
+      return `<p>Households in ${oa} have an average of ${value} cars</p>`;
+    } else {
+      return `<p>There are ${value.toLocaleString()} people per square kilometre in ${oa}</p>`;
+    }
+  }
+
+  function onClick(e: CustomEvent<MapGeoJSONFeature>) {
+    let oa = e.detail.properties["OA21CD"];
+    if (
+      colorBy == "percent_households_without_car" ||
+      colorBy == "average_cars_per_household"
+    ) {
+      window.open(
+        `https://www.ons.gov.uk/census/maps/choropleth/housing/number-of-cars-or-vans/number-of-cars-5a/no-cars-or-vans-in-household?oa=${oa}`,
+        "_blank"
+      );
+    } else {
+      window.open(
+        `https://www.ons.gov.uk/census/maps/choropleth/population/population-density/population-density/persons-per-square-kilometre?oa=${oa}`,
+        "_blank"
+      );
+    }
+  }
+
+  // Should return 6 things (1 more than colorScale)
+  function makeLimits(): number[] {
```

Comment:
Chris, I still need to fix my Python dependencies to run your script and see what the cut-offs for the bandings you used. I tried out different quantile functions in pandas, but they don't seem to handle the min / max value. For MapLibre, we have to set a real max value, or it uses a fallback color. So I've made some simple first choices here, but happy to iterate on them